app/models/concerns/digital_object_concerns/attribute_assignment.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module DigitalObjectConcerns
+  module AttributeAssignment
+    include DigitalObjectConcerns::AttributeAssignment::Doi
+    include DigitalObjectConcerns::AttributeAssignment::DynamicFieldData
+    include DigitalObjectConcerns::AttributeAssignment::Identifiers
+    include DigitalObjectConcerns::AttributeAssignment::OptimisticLockToken
+    include DigitalObjectConcerns::AttributeAssignment::ParentUids
+    include DigitalObjectConcerns::AttributeAssignment::Preserve
+    include DigitalObjectConcerns::AttributeAssignment::Projects
+    include DigitalObjectConcerns::AttributeAssignment::PendingPublishEntries
+    include DigitalObjectConcerns::AttributeAssignment::Resources
+    include DigitalObjectConcerns::AttributeAssignment::Rights
+    include DigitalObjectConcerns::AttributeAssignment::State
+    include DigitalObjectConcerns::AttributeAssignment::StructuredChildren
+
+    # A powerful method that can set many of this object's properties in one go, based on the given digital_object_data hash.
+    # This method will raise errors if it is given invalid data (e.g. references to projects or publish targets that don't exist),
+    # so be ready to handle those exceptions. All of the deliberately-thrown exceptions will extend Hyacinth::Exceptions::HyacinthError.
+    # @param digital_object_data [Hash] A hash of data used to update many of this object's properties.
+    # @param merge_dynamic_fields [boolean] If true, merges given dynamic_field_data Hash into into existing dynamic_field_data.
+    #        If false, replaces existing dynamic_field_data with new dynamic_field_data Hash.
+    def assign_attributes(new_digital_object_data, merge_dynamic_fields)
```

Comment:
Would it be bad to have a default value for `merge_dynamic_fields`?

Reply:
could we maybe make it a named param and force one to be provided if `dynamic_field_data` is present? It just seems weird to provide the setting when we aren't setting `dynamic_field_data`. We can also make this a ticket and implement later.